From 25bf770527fead68488db8f2418e41eb7a7b0b95 Mon Sep 17 00:00:00 2001
From: Ilya Alekseyev <ilya.alekseyev@nutanix.com>
Date: Wed, 11 Oct 2023 22:07:22 -0400
Subject: [PATCH 6/8] Nutanix improvements

- Fetch Nutanix RHEL source image URL from environment
- Force-delete Nutanix builder VMs on failure

Signed-off-by: Vignesh Goutham Ganesh <vgg@amazon.com>
---
 images/capi/packer/nutanix/packer.json | 3 ++-
 images/capi/packer/nutanix/rhel-8.json | 1 -
 images/capi/packer/nutanix/rhel-9.json | 1 -
 3 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/images/capi/packer/nutanix/packer.json b/images/capi/packer/nutanix/packer.json
index 2d88c2fb6..f46c8d8d2 100644
--- a/images/capi/packer/nutanix/packer.json
+++ b/images/capi/packer/nutanix/packer.json
@@ -132,6 +132,7 @@
     "image_delete": "false",
     "image_export": "false",
     "image_name": "{{user `build_name`}}-kube-{{user `kubernetes_semver`}}",
+    "image_url": "{{env `RHEL_IMAGE_URL`}}",
     "kubernetes_cni_deb_version": null,
     "kubernetes_cni_http_source": null,
     "kubernetes_cni_semver": null,
@@ -164,6 +165,6 @@
     "source_image_force": "false",
     "ssh_password": "builder",
     "ssh_username": "builder",
-    "vm_force_delete": "false"
+    "vm_force_delete": "true"
   }
 }
diff --git a/images/capi/packer/nutanix/rhel-8.json b/images/capi/packer/nutanix/rhel-8.json
index 9aba21d66..b2bef7674 100644
--- a/images/capi/packer/nutanix/rhel-8.json
+++ b/images/capi/packer/nutanix/rhel-8.json
@@ -5,7 +5,6 @@
   "epel_rpm_gpg_key": "https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-8",
   "extra_rpms": "",
   "guest_os_type": "Linux",
-  "image_url": "https://REPLACE_YOUR_SERVER/redhat/8/rhel-8.8-x86_64-kvm.qcow2",
   "redhat_epel_rpm": "https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm",
   "shutdown_command": "shutdown -P now",
   "user_data": "I2Nsb3VkLWNvbmZpZwp1c2VyczoKICAtIG5hbWU6IGJ1aWxkZXIKICAgIHN1ZG86IFsnQUxMPShBTEwpIE5PUEFTU1dEOkFMTCddCmNocGFzc3dkOgogIGxpc3Q6IHwKICAgIGJ1aWxkZXI6YnVpbGRlcgogIGV4cGlyZTogRmFsc2UKc3NoX3B3YXV0aDogVHJ1ZQ=="
diff --git a/images/capi/packer/nutanix/rhel-9.json b/images/capi/packer/nutanix/rhel-9.json
index b7dddb4f2..921a9729f 100644
--- a/images/capi/packer/nutanix/rhel-9.json
+++ b/images/capi/packer/nutanix/rhel-9.json
@@ -5,7 +5,6 @@
   "epel_rpm_gpg_key": "https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-9",
   "extra_rpms": "",
   "guest_os_type": "Linux",
-  "image_url": "https://REPLACE_YOUR_SERVER/redhat/9/rhel-9.2-x86_64-kvm.qcow2",
   "redhat_epel_rpm": "https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm",
   "shutdown_command": "shutdown -P now",
   "user_data": "I2Nsb3VkLWNvbmZpZwp1c2VyczoKICAtIG5hbWU6IGJ1aWxkZXIKICAgIHN1ZG86IFsnQUxMPShBTEwpIE5PUEFTU1dEOkFMTCddCmNocGFzc3dkOgogIGxpc3Q6IHwKICAgIGJ1aWxkZXI6YnVpbGRlcgogIGV4cGlyZTogRmFsc2UKc3NoX3B3YXV0aDogVHJ1ZQ=="
-- 
2.45.2

